scripts/managedb/mdsvector2mdsobject.py

```diff
@@ -0,0 +1,226 @@
+#!/usr/bin/python
+# -*- coding: latin-1 -*-
+# Copyright 2016 Telefonica Investigacion y Desarrollo, S.A.U
+#
+# This file is part of Orion Context Broker.
+#
+# Orion Context Broker is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# Orion Context Broker is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+#
+# For those usages not covered by this license please contact with
+# iot_support at tid dot es
+
+
+# This script is inspired in old attrsvector2attrsobject.py
+
+__author__ = 'fermin'
+
+from pymongo import MongoClient
+import json
+import sys
+from time import sleep
+
+ATTRS     = 'attrs'
+MDS       = 'md'
+MD_NAMES  = 'mdNames'
+
+# The way in which Python manage dictionaries doesn't make easy to be sure
+# of field ordering, which is important for MongoDB in the case of using an
+# embedded document for _id. The flatten() function helps
+def flatten(_id):
+    r = {'_id.id': _id['id']}
+
+    if 'type' in _id:
+       r['_id.type'] = _id['type']
+    else:
+       r['_id.type'] = {'$exists': False}
+
+    if 'servicePath' in _id:
+       r['_id.servicePath'] = _id['servicePath']
+    else:
+       r['_id.servicePath'] = {'$exists': False}
+
+    return r
+
+def update_ok(doc, n_attrs, n_sum_md_names, n_sum_mds):
+
+    if (not ATTRS in doc):
+        print "debug1"
+        return False
+
+    ck_attrs = len(doc[ATTRS].keys())
+
+    ck_sum_md_names = 0
+    ck_sum_mds      = 0 
+    for attr in doc[ATTRS].keys():
+
+        # By construction, mdNames is always there (sometimes empty), but mds may be or not
+        md_names = doc[ATTRS][attr][MD_NAMES]
+        ck_sum_md_names += len(md_names)
+
+        if MDS in doc[ATTRS][attr].keys():        
+            mds         = doc[ATTRS][attr][MDS]            
+            ck_sum_mds  += len(mds.keys())
+
+
+    if (n_attrs != ck_attrs or n_sum_md_names != ck_sum_md_names or n_sum_mds != ck_sum_mds):
+        print "debug 2: %d %d %d %d %d %d" % (ck_attrs, n_attrs, n_sum_md_names, ck_sum_md_names, n_sum_mds, ck_sum_mds)
+        return False
+
+    return True
+
+if len(sys.argv) != 2:
+    print "missing db name"
```

Comment:
Actually I'm not talking about a better parseArgs here.
Just a different, more accurate print text in case the script is called with too many/few parameters